task/backend/scheduler/scheduler.go

```diff
@@ -0,0 +1,78 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb"
+)
+
+type Storage interface {
+	// UpdateLatestCompleted will allow us to set a latest completed time for an id.
+	UpdateLatestCompleted(ctx context.Context, id influxdb.ID, lastRun time.Time) error
+
+	// retrieve the latest completed for this schedulable id.
+	// If GetLatestCompleted is empty scheduler will use "now" as its latest completed
+	GetLatestCompleted(ctx context.Context, id influxdb.ID) (time.Time, error)
+}
+
+type Executor interface {
+	// Execute is used to execute run's for any schedulable object.
+	// the executor can go through manual runs, clean currently running, and then create a new run based on `now`.
+	// if Now is zero we can just do the first 2 steps (This is how we would trigger manual runs).
+	// Errors returned from the execute request imply that this attempt has failed and
+	// should be put back in scheduler and re executed at a alter time. We will add scheduler specific errors
+	// so the time can be configurable.
+	Execute(ctx context.Context, id influxdb.ID, meta interface{}, now time.Time) (Execution, error)
```

Comment:
Is there any way to give `meta` more structure?  It always feels too free-form to me that an interface defines a function that has an interface argument.

Reply:
I had thought about that, I dont love that it can be anything. I wanted it to be agnostic to our usage. I worried that by defining what it would look like may restrict its usefulness as a open source package.